source/hardware/controllers/hercules_djcontrol_inpulse_300.rst

```diff
@@ -54,58 +54,120 @@ Controls not included in this mapping
+
+Roll Mode
+^^^^^^^^^
+
+===============================================================  ==========================================
+Control                                                          Function
+===============================================================  ==========================================
+Pad 1                                                            Activate Beatloop of 1/8 beat size.
+Pad 2                                                            Activate Beatloop of 1/4 beat size.
+Pad 3                                                            Activate Beatloop of 1/2 beat size.
+Pad 4                                                            Activate Beatloop of 1 beat size.
+Pad 5                                                            Activate Beatloop of 2 beat size.
+Pad 6                                                            Activate Beatloop of 4 beat size.
+Pad 7                                                            Activate Beatloop of 8 beat size.
+Pad 8                                                            Activate Beatloop of 16 beat size.
+===============================================================  ==========================================
+
+FX Mode
+^^^^^^^
+
+===============================================================  ==========================================
+Control                                                          Function
+===============================================================  ==========================================
+Pad 1 - 3                                                        Toggle Effect 1 - 3.
+Pad 4 (deck A)                                                   Toggle Effect Rack 1 for Deck A.
+Pad 4 (deck B)                                                   Toggle Effect Rack 2 for Deck B.
+Shift + Pad 4 (deck A)                                           Toggle Effect Rack 1 for Deck B.
+Shift + Pad 4 (deck B)                                           Toggle Effect Rack 1 for Deck A.
+===============================================================  ==========================================
 
 .. note::
    When using **Firmware v1.68** and earlier, each FX pad will
-   send multiple and different Note and CC messages. As these could not all
-   be used properly with Mixxx current effect framework, the pads have not
-   been assigned in this mapping. \* However, after applying **Firmware
-   v1.72**, each FX pad now sends a simple Note On/Note Off (as is the case
-   in every other Pad mode). This would make it easier to use the pads to
-   control effect and the mapping might be modified eventually to take
-   advantage of this possibility.
+   send multiple and different Note and CC messages. It is recommended to apply **Firmware
+   v1.72** update, that will patch each FX pad to send a simple Note On/Note Off (as is the case
```

Comment:
So the mapping only works correctly with firmware 1.72 or later? I suggest rephrase this to make it clear that users should upgrade and move this note into the compatibility section.

Reply:
Well.. it works with `SHIFT`+Pad without the upgrade but for the functionality without `SHIFT` it requires upgrade. I've bought my Inpulse just last month and it was already upgraded to the latest firmware.

Do you have a suggestion for the rewording?

Reply:
I've tried something
cd53b3e

Reply:
Thanks. 

I still think that this note is confusing to regular users that are not familiar with the MIDI spec and don't know what "cc messages" are.

I suggest something concise that also mentions what happens if you don't upgrade:

> Please upgrade the controller firmware to 1.72 or later. With older firmware versions, pressing `SHIFT` + Pad will do nothing.

The technical details could be added as a comment to the source instead.

Reply:
Moved it into Compatibility section
09934aa 

__With older firmware the pads will only work with `SHIFT` + Pad__